e2e-tests/static/main.js

```diff
@@ -0,0 +1,82 @@
+class EventReceiver {
+    constructor(endpoint, messagesId) {
+        this.source = new EventSource(endpoint);
+        this.source.onmessage = this.receiveEvent.bind(this);
+        this.messagesId = messagesId;
+    }
+
+    receiveEvent(event) {
+        let newElement = document.createElement('div');
+        newElement.classList.add('message');
+        newElement.classList.add('sse');
+        newElement.textContent = event.data;
```

Comment:
shouldn't we guard form an empty message here? (if `yield None` was sent by the emitter)

Reply:
No, these seem to be some sort of keepalive thing, they are not surfaced as events on the front-end side.

See also [`Response.sse`](https://falcon.readthedocs.io/en/latest/api/request_and_response_asgi.html#falcon.asgi.Response.sse):
```python
        if not some_event:
            # Send an event consisting of a single "ping"
            #   comment to keep the connection alive.
            yield SSEvent()

            # Alternatively, one can simply yield None and
            #   a "ping" will also be sent as above.

            # yield

            continue
```

Reply:
Oh ok. I looked only at mdn and it was not mentioned that messages with empty body are discarded, but looking at the linker speck it does seems so. From [Interpreting an event stream](https://html.spec.whatwg.org/multipage/server-sent-events.html#dispatchMessage):

> If the data buffer is an empty string, set the data buffer and the event type buffer to the empty string and return.